src/iconv.jl

```diff
@@ -1,9 +1,54 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
 module iconv
-import Base: close, eof, flush, read, readall, write
+import Base: close, eof, flush, read, readall, write, show
 import Base.Libc: errno, strerror, E2BIG, EINVAL, EILSEQ
 export StringEncoder, StringDecoder, encode, decode
+export StringEncodingError, OutputBufferError, IConvError
+export InvalidEncodingError, InvalidSequenceError, IncompleteSequenceError
+
+
+abstract StringEncodingError
+
+# specified encodings are not supported
+type InvalidEncodingError <: StringEncodingError
+    args::Tuple{ASCIIString, ASCIIString}
+    InvalidEncodingError(from, to) = new((ascii(from), ascii(to)))
```

Comment:
I mean, you could make these outer constructors, and I'm not sure the `ascii` call is needed.